app/services/clients/stripe/stripeCompany.model.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+// NPM dependencies
+const Joi = require('joi')
+
+const schema = {
+  vat_id: Joi.string().required(),
+  tax_id: Joi.string().optional()
+}
+
+class StripeCompany {
+  constructor (body) {
+    const params = Object.assign({}, body)
+    const { error, value: model } = Joi.validate(params, schema)
+
+    if (error) {
+      throw new Error(`StripeCompany ${error.details[0].message}`)
+    }
+
+    Object.assign(this, build(model))
+  }
+
+  basicObject () {
+    return Object.assign({}, this)
+  }
+}
+
+function build (params) {
+  const stripeCompany = {
+    company: {
+      vat_id: params.vat_id.replace(/\W/g, '')
```

Comment:
Not sure we should be doing this normalisation here. I think the controller that takes the input should probably do it and this class should just require the input to be well-formed.

Reply:
Sure, I removed the normalisation from here if it will be handled in controller, thanks 👍